src/main/java/httpserver/HTTPHandler.java

```diff
@@ -22,232 +22,187 @@
-      String path = MethodWrapper.cleanPath(request.getPath());
-      MethodWrapper method = map.get(path);
-  
-      /*  If the above MethodWrapper is null (occurs when the requested path
-          is dynamic), find the best fit method based on MethodWrapper's scoring
-          technique (see MethodWrapper#howCorrect for more information).
-       */
-      if (method == null) {
-        int bestFit = 0;
-        Set<String> keys = map.keySet();
-        for (String key : keys) {
-          MethodWrapper testMethod = map.get(key);
-          int testScore = testMethod.howCorrect(path);
-  
-          if (testScore > bestFit) {
-            method = testMethod;
-            bestFit = testScore;
-          }
-        }
-      }
-  
-      /*  If none of the paths match the request's path, try using a wild-card
-          or root path in that order.
-       */
-      if (method == null) {
-        if (map.containsKey("*")) {
-          method = map.get("*");
-        }
-      }
-  
-      /*  If, following the whole ordeal, no acceptable method is found, send the
-          client a 501, Not a Method error.
-       */
-      if (method == null) {
-        response.message(501, HTTPResponse.NOT_A_METHOD_ERROR);
-        return;
-      }
-  
-      System.out.println("Method Invoked: " + method);
-      method.invoke(this, response, request, path);
+    private final HashMap<String, ArrayList<MethodWrapper>> routes = new HashMap<>();
+
+    private Socket socket;
+    private DataOutputStream writer;
+
+
+    /**
+     * Create an HTTPHandler. <p>
+     *
+     * When writing your own HTTPHandler, this is where you should add the
+     * handler's internal routing, as well performing any setup tasks. Handlers
+     * are multi-use, which means that only one of any kind of handler should be
+     * created in an application (unless you have custom needs).
+     *
+     * @throws HTTPException  The exception typically comes from trying to add
+     *                        a new method. In a standard configuration this will
+     *                        keep the server from starting.
+     */
+    public HTTPHandler() throws HTTPException { }
+
+
+    /**
+     * Where the Handler handles the information given from the request and
+     * based off of the paths specified in the Handler. <p>
+     *
+     * This can be overridden for more fine-grained handling. As is, it uses
+     * the data behind the addGET, addPOST, and addDELETE methods for determining
+     * the correct action to take. <p>
+     *
+     * If there is not exact match, the `*` path is used. If you don't have a `*`
+     * catchall route, a 501 (Not implemented) is sent to the client.
+     *
+     * @param request     The incoming HTTPRequest.
+     * @param response    The outgoing HTTPResponse, waiting to be filled by an
+     *                    HTTPHandler.
+     *
+     * @see HTTPHandler#addGET
+     * @see HTTPHandler#addPOST
+     * @see HTTPHandler#addDELETE
```

Comment:
`addGET`, `addPOST`, and `addDELETE` have been renamed `get`, `post`, and `delete`.